crates/polars-core/src/frame/mod.rs

```diff
@@ -178,6 +179,10 @@ pub struct DataFrame {
     height: usize,
     // invariant: columns[i].len() == height for each 0 >= i > columns.len()
     pub(crate) columns: Vec<Column>,
+
+    /// A cached schema. This might not give correct results if the DataFrame was modified in place
+    /// between schema and reading.
+    cached_schema: OnceLock<SchemaRef>,
```

Comment:
I don't think we should add a schema we cannot trust. That feel very brittle.

Reply:
Mmmh, I personally think this is fine. I think the alternative -- updating a schema to keep it correct -- is probably more error-prone.

Reply:
> -- updating a schema to keep it correct -- is probably more error-prone.

Yeah, that's also tricky. Alright.